Add 'dot' line mode

ybhvf opened the pull request:
This commit adds a new 'dot' line mode which produces small dots via very short lines. This is useful for plotting dithered art.

ybhvf committed:
Add 'dot' line mode

This commit adds a new 'dot' line mode which produces small dots via
very short lines. This is useful for plotting dithered art.

abey79 commented:
Thanks for your contribution!

This is an interesting mode, however I don't understand the rationale behind using the `PIXEL_OFFSET` multiplier. This offset of 5 is related to the `big` mode, where each pixel is made of a 5x5 pen widths square. If the intent is indeed to spread out the points, it probably should be a command argument.

Also:
- Tiny circles of diameter ~1-2 tenth of pen widths yield better result than small lines with most pen. I think it would be better for this mode (and I should likely do it every time a single pixel is drawn in the other modes btw).
- Do you have any examples to contribute for the doc?

ybhvf committed:
Use circular paths for dot mode

ybhvf commented:
Hah, there's no reason behind my using the `PIXEL_OFFSET` multiplier... It's leftover from `big_mode()`, which I copied and modified for `dot_mode()`.

I've removed `PIXEL_OFFSET` from `dot_mode()`, and have updated it to create circle paths instead. I'll also add an example later today :)

abey79 commented on pixelart/pixelart.py:
```diff
@@ -149,25 +149,38 @@ def snake_mode(document: vp.Document, img: np.ndarray, colors: np.ndarray, pen_w
+        )
+
+        paths = [vp.circle(i, j, pen_width * 0.05)
+                 for i, j in zip(indice_j, indice_i)]
```
This should read as follows, for correctness and clarity:
```
paths = [vp.circle(j * pen_width, i * pen_width, pen_width * 0.05)
                  for j, i in zip(indice_j, indice_i)]
```

In particular, the `* pen_width` is needed such that the grid is indeed scaled by the pen width.

abey79 commented:
FYI, I made some changes that will probably need you to rebase and force-push.
